react-components/src/components/RuleBasedOutputs/RuleBasedOutputs.tsx

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { useEffect, type ReactElement } from 'react';
+
+import { type CogniteCadModel } from '@cognite/reveal';
+import { useAllMappedEquipmentAssetMappings, useReveal } from '../..';
+import { Color } from 'three';
+import { type RuleOutputSet } from './types';
+import { generateRuleBasedOutputs } from './utils';
+
+export type ColorOverlayProps = {
+  ruleSet: RuleOutputSet | undefined;
+};
+
+export function RuleBasedOutputs({ ruleSet }: ColorOverlayProps): ReactElement | undefined {
+  const viewer = useReveal();
+
+  // only enabled whether has loaded models
+  if (viewer.models === undefined || viewer.models.length === 0) return;
```

Comment:
We cannot have an early return when there are React hooks later in the component. The hooks will just have to tackle the case when `models` are undefined on their own

Reply:
fixed here

https://github.com/cognitedata/reveal/pull/4210/commits/e4c1637a2c6f4412f8fa79861d3d37844a646860